### DISCUSSION TITLE
my app isn't deploying correctly

### DISCUSSION THREAD

**Discussion by rayzhou4** [vercel/community#7208] [Help] [answered]
### Summary

Hi! 

**MAIN ISSUE:**
I have this issue where I am trying to deploy the newest version of my react app, but when i open the app, it seems to always revert back to my old code source. 

**EXPLANTION:**
So basically, I am using Spotify API (idk if this affects it or not) and I call a function that supposed authenticates the user and then the callback function is called and that when we return back from Spotify's authorization page to my website. But when I am checking my network tab in the dev's tools, i can see that whenever it is return back from the callback, it shows my old code source??

Things I have already tried:
- redeploying
- making sure my github code is up to date
- checking that the code source of the vercel build is correct (i checked it on vercel's dashboard and it is actually correct)
- trying to run it locally (it works fine)

someone please help. 😭

### Steps to Reproduce

if you simply click on my website deployment [here](https://spotifyguesser-io.vercel.app/), i have tried looking through dev tool's network tab, and whenever i click it and try to authenticate myself, it always shows up these two token calls (which used to be in my old code source) and instead, it should be calling the third picture's token call's api endpoint which is correct. 

these first two pics are from the buggy vercel deployment (the home.js and app.js files dont have these api calls, but when i click into, it shows my old code, and my old code did use to call the '/auth/token' endpoint)
![image](https://github.com/vercel/community/assets/85908460/4a26b054-db61-43b4-b237-99311dc927c7)
![image](https://github.com/vercel/community/assets/85908460/412936b0-1aa6-4869-b8d7-872c4df97cd1)

this picture is how its supposed to go
![image](https://github.com/vercel/community/assets/85908460/12e7629c-497a-477a-bd70-c5dd0054223d)

**Comment by Pozi09** [comment]
Hi. I can't help you with your problem, but I can help you with promoting your app.

**Reply by rayzhou4** [reply]
what do you mean by promoting? as in getting it into production?

**Reply by Pozi09** [reply]
Promote on social networks and monetize it

**Comment by versecafe** [comment]
@rayzhou4 Can you make sure to clear out your build and Data cache?

**Reply by rayzhou4** [reply]
how do i do that?

**Reply by versecafe** [reply, marked as answer]
make sure to run a rebuild and in the menu make sure build cache is off and then go to `https://vercel.com/[team]/[project]/settings/data-cache` and clear it, I'm not sure what else could cause this except maybe a messed up Skew Protection setting `https://vercel.com/[team]/[project]/settings/advanced` and check if Skew Protection is on

**Reply by rayzhou4** [reply]
thank you! it works!!

[a comment by an automated account is not shown]
